### PR TITLE
Docs v4: Sass implementation and rounding precision

### PR DESCRIPTION
**Pull request by voltaek**
@XhmikosR Here is the backport you requested in #32512.

This makes changes similar to those made to the v5 docs to update and clarify the Sass compiler used and the minimum rounding precision we recommend with it. The details added are linked to from several places now instead of bits of the info being spread throughout the docs.

I added some notes about Node Sass being deprecated and how Dart Sass should be a mostly drop in replacement if the user would need support for newer Sass features or CSS standards. It's still made clear that we build Bootstrap v4 with Node Sass, so I think this is a good balance of telling them it's still alright to compile with it while also giving them options if they need newer features.

### CODE REVIEW

File: site/content/docs/4.5/getting-started/build-tools.md [hunk 1]
```diff
@@ -30,6 +30,14 @@ Our [package.json]({{< param repo >}}/blob/v{{< param current_version >}}/packag
 
 Run `npm run` to see all the npm scripts.
 
+## Sass
+
+Bootstrap v4 uses [Node Sass](https://github.com/sass/node-sass) for compiling our Sass source files into CSS files (included in our build process). In order to end up with the same generated CSS when compiling Sass using your own asset pipeline, you'll need to use a Sass compiler that supports at least the features that Node Sass does. This is important to note because as of October 26, 2020, LibSass and packages built on top of it—including Node Sass—are [deprecated](https://sass-lang.com/blog/libsass-is-deprecated).
+
+If you require newer Sass features or compatibility with newer CSS standards, [Dart Sass](https://sass-lang.com/dart-sass) is now the primary implementation of Sass and supports a JavaScript API that's fully compatible with Node Sass (with a few exceptions listed on both the project's [GitHub](https://github.com/sass/dart-sass#javascript-api) or [npm](https://www.npmjs.com/package/sass#user-content-api) pages).
```
Comment (XhmikosR) [inline site/content/docs/4.5/getting-started/build-tools.md:37]: I'd change this to link to the GitHub repo only

Comment (voltaek) [reply site/content/docs/4.5/getting-started/build-tools.md]: Done

**Comment by voltaek** [comment]
Is this slated for v4.6.0 or will it have to wait until v4.6.1?

**Comment by voltaek** [comment]
Rebased against `v4-dev` and resolved conflict with mdo's new callout in `build-tools.md`

**Review by mdo — APPROVED** [review]

**Pull request merged**